pkgs/swiftgen/test/integration/util.dart

```diff
@@ -0,0 +1,166 @@
+
+  TestGenerator(this.name) {
+    testDir = path.absolute(path.join(pkgDir, 'test/integration'));
+    tempDir = path.join(testDir, 'temp');
+    inputFile = path.join(testDir, '${name}_wrapper.swift');
+    outputFile = path.join(tempDir, '${name}_output.dart');
+    outputObjCFile = path.join(tempDir, '${name}_output.m');
+    objInputFile = path.join(tempDir, '$name.o');
+    objWrapperFile = path.join(tempDir, '${name}_wrapper.o');
+    objObjCFile = path.join(tempDir, '${name}_output_m.o');
+    dylibFile = path.join(tempDir, '$name.dylib');
+    actualOutputFile = path.join(testDir, '${name}_bindings.dart');
+  }
+
+  Future<void> generateBindings() async => SwiftGen(
+    target: await Target.host(),
+    input: ObjCCompatibleSwiftFileInput(
+      module: name,
+      files: [Uri.file(inputFile)],
+    ),
+    tempDirectory: Directory(tempDir).uri,
+    ffigen: fg.Config(
+      output: Uri.file(outputFile),
+      outputObjC: Uri.file(outputObjCFile),
+      objcInterfaces: fg.DeclarationFilters(
+        shouldInclude: (decl) => decl.originalName.startsWith('Test'),
+      ),
+      preamble: '''
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: always_specify_types
+// ignore_for_file: camel_case_types
+// ignore_for_file: non_constant_identifier_names
+// ignore_for_file: unnecessary_non_null_assertion
+// ignore_for_file: unused_element
+// ignore_for_file: unused_field
+// coverage:ignore-file
+''',
+    ),
+  ).generate();
+
+  Future<void> generateAndVerifyBindings() async {
+    // Run the generation pipeline. This produces the swift compatability
+    // wrapper, and the ffigen wrapper.
+    await generateBindings();
+
+    expect(File(inputFile).existsSync(), isTrue);
+    expect(File(outputFile).existsSync(), isTrue);
+    expect(File(outputObjCFile).existsSync(), isTrue);
+
+    // The generation pipeline also creates some obj files as a byproduct.
+    expect(File(objWrapperFile).existsSync(), isTrue);
+
+    // We also need to compile outputObjCFile to an obj file.
+    await run('clang', [
+      '-x',
+      'objective-c',
+      '-fobjc-arc',
+      '-c',
+      outputObjCFile,
+      '-fpic',
+      '-o',
+      objObjCFile,
+    ], tempDir);
+    expect(File(objObjCFile).existsSync(), isTrue);
+
+    // Link all the obj files into a dylib.
+    await run('clang', [
+      '-shared',
+      '-framework',
+      'Foundation',
+      objWrapperFile,
+      objObjCFile,
+      '-o',
+      dylibFile,
+    ], tempDir);
+    expect(File(dylibFile).existsSync(), isTrue);
+  }
+}
+
+/// Test files are run in a variety of ways, find this package root in all.
+///
+/// Test files can be run from source from any working directory. The Dart SDK
+/// `tools/test.py` runs them from the root of the SDK for example.
```

Comment:
Do you want to run these tests in the Dart SDK CI?

Reply:
No. I've just been copy/pasting the whole util. We should add a helper package to this monorepo so we can share utils like this.